_posts/2018-12-04-imposing-invariants-in-scala.md

```diff
@@ -0,0 +1,359 @@
+---
+layout: post
+author: cmteixeira
+title: Imposing invariants in Scala
+---
+
+
+
+Imagine a opera house allows reservations to a show. For each reservation two people can come: The buyer of the reservation himself, and his guest.  
+As in any show-house, there are many types of seats; some are standing, some normal and some on vip tribunes.
+
+
+
+As a restriction from the opera house, both buyer and guest must be on the same seating area. For example, it is not permitted for one ticket to be standing while the other for the normal seated area.  
+I believe this is what people call an invariant. A requirement that must hold for the data to have meaning.  
+How to enforce then this constraint ?  
```

Comment:
Please remove the extra whitespace before the question mark.

Reply:
done